lib/galaxy/datatypes/text.py

```diff
@@ -1532,3 +1532,55 @@ def sniff_prefix(self, file_prefix: FilePrefix) -> bool:
         False
         """
         return file_prefix.startswith("RBT_PARAMETER_FILE_V1.00")
+
+
+class ODV(Text):
+    file_ext = "odv"
+
+    MetadataElement(name="meta_var_count", default=0, desc="Number of Meta Variables", readonly=True)
+    MetadataElement(name="var_count", default=0, desc="Number of Variables", readonly=True)
+
+    def sniff_prefix(self, file_prefix: FilePrefix) -> bool:
+        """
+        Determines whether the file is has some required details in General section
+        """
+        header = file_prefix.contents_header.lower()
+        keywords = ["collectionformat", "odvcf"]
+        return all(keyword in header for keyword in keywords)
+
+    def set_meta(self, dataset: DatasetProtocol, overwrite: bool = True, **kwd):
+        dataset.metadata.meta_var_count = 0
+        dataset.metadata.var_count = 0
+
+        section = None
+        try:
+            with open(dataset.get_file_name(), encoding="utf-8") as f:
```

Comment:
This potentially iterates over the whole file, with potentially unlimited memory consumption on long lines.

- Do you need to read the whole file or is something like the first MB fine ?
- Do you actually make use of `meta_var_count` and `var_count` or can that be added later ?